test/date-time/date-time-properties.test.ts

```diff
@@ -0,0 +1,259 @@
+    {
+        test(`Given a luxon date time
+        when DateTime is created from that value and zone
+        then both should have same timeStamp`,
+            () =>
+            {
+                const timeStamp = LuxonDateTime.fromFormat("2024-01-01 10:00", "yyyy-MM-dd HH:mm", { zone: "utc" }).toSeconds();
+                Assert.strictEqual(new DateTime({ value: "2024-01-01 10:00", zone: "utc" }).timestamp, timeStamp);
+            });
+
+        test(`Given a value epoch start ("1970-01-01 00:00") in utc
+        when DateTime is created from that
+        then it should have timeStamp 0`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "1970-01-01 00:00", zone: "utc" }).timestamp, 0);
+            }
+        );
+
+        test(`Given a value before epoch start ("1969-12-31 23:59") in utc
+        when DateTime is created from that
+        then it should have timeStamp negative (-60)`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "1969-12-31 23:59", zone: "utc" }).timestamp, -60);
+            }
+        );
+
+        test(`Given a value epoch start ("1970-01-01 00:01") in utc
+        when DateTime is created from that
+        then it should have timeStamp positive (60)`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "1970-01-01 00:01", zone: "utc" }).timestamp, 60);
+            }
+        );
+    });
+
+
+    suite("Date code", () =>
+    {
+        test(`Given a date time value (2024-01-01 10:00)
+        when DateTime is created from that value
+        then date code should be 20240101`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "2024-01-01 10:00", zone: "utc" }).dateCode, "20240101");
+            }
+        );
+    });
+
+    suite("Time code", () =>
+    {
+        test(`Given a date time value (2024-01-01 10:00)
+        when DateTime is created from that value
+        then time code should be 1000`,
+            () =>
+            {
+                Assert.strictEqual(new DateTime({ value: "2024-01-01 10:00", zone: "utc" }).timeCode, "1000");
+            }
+        );
+    });
+
+    suite("Date value", () =>
+    {
+        const value = "2024-01-01 10:00";
+        const dateTime = new DateTime({ value, zone: "utc" });
+        const dateValue = value.split(" ").takeFirst();
+
+        test(`Given a valid value (${value}) 
+        when a DateTime is created from that value
+        then date value should be ${dateValue}`,
+            () =>
+            {
+                Assert.strictEqual(dateTime.dateValue, dateValue);
+            }
+        );
+
+        test(`Given a valid value (${value}) 
+        when a DateTime is created from that value
+        then the date value property should be in valid date format`,
+            () =>
+            {
+                Assert.ok(DateTime.validateDateFormat(dateTime.dateValue));
```

Comment:
you are validating one functionality with other functionality of the same thing, this test will fail if either of the 2 are messed up

Reply:
you can use `matchesFormat` to validate format

Reply:
or you can just check you get the expected value from dateTime.dateValue